### ISSUE TITLE
RDBInterrupt.InsertOp guarauntee failure in release mode

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4038]
On v2.0.x when compiling in release mode (`make UNIT_TESTS=1`):
```
daniel@newton:~/rethinkdb/build/release_clang$ ./rethinkdb-unittest --gtest_filter=RDBInter*
Note: Google Test filter = RDBInter*
[==========] Running 5 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 5 tests from RDBInterrupt
[ RUN      ] RDBInterrupt.InsertOp
Version: rethinkdb foo (CLANG 3.2 (tags/RELEASE_32/final))
error: Error in src/unittest/rdb_interruptor.cc at line 81:
Previous message may not have been written to the log file (cannot open or find log file).
error: Guarantee failed: [*count_out > 0] 
Previous message may not have been written to the log file (cannot open or find log file).
error: Backtrace:
Previous message may not have been written to the log file (cannot open or find log file).
error: Thu Apr  9 13:34:30 2015
       
       1: backtrace_t::backtrace_t() at backtrace.cc:203
       2: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at basic_string.h:269
       3: format_backtrace(bool) at backtrace.cc:197
       4: report_fatal_error(char const*, int, char const*, ...) at basic_string.h:287
       5: unittest::count_evals(unittest::test_rdb_env_t*, ql::protob_t<Term const>, unsigned int*, unittest::verify_callback_t*) at rdb_interruptor.cc:81
       6: std::_Function_handler<void (), std::_Bind<void (*(unittest::test_rdb_env_t*, ql::protob_t<Term const>, unsigned int*, unittest::exists_verify_callback_t*))(unittest::test_rdb_env_t*, ql::protob_t<Term const>, unsigned int*, unittest::verify_callback_t*)> >::_M_invoke(std::_Any_data const&) at counted.hpp:60
       7: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       8: coro_t::run() at coroutines.cc:214
Previous message may not have been written to the log file (cannot open or find log file).
error: Exiting.
Previous message may not have been written to the log file (cannot open or find log file).
Trace/breakpoint trap (core dumped)
```

It might just be a problem with the test in release mode (it works fine in debug mode), but we should check.

@larkost have you seen this problem before?

**Comment by danielmewes** [comment]
Looks like the callback that this test is relying on isn't getting called in release mode:
```
std::vector<datum_t>
datum_stream_t::next_batch(env_t *env, const batchspec_t &batchspec) {
    DEBUG_ONLY_CODE(env->do_eval_callback());
```

**Comment by anatol** [comment]
I confirm this test failure. On Arch I see

```
[ RUN      ] RDBInterrupt.InsertOp
Version: rethinkdb 2.0.0-1 (GCC 4.9.2)
error: Error in src/unittest/rdb_interruptor.cc at line 81:
Previous message may not have been written to the log file (cannot open or find log file).
error: Guarantee failed: [*count_out > 0] 
Previous message may not have been written to the log file (cannot open or find log file).
error: Backtrace:
Previous message may not have been written to the log file (cannot open or find log file).
error: Tue Apr 14 15:00:13 2015
       
       1: backtrace_t::backtrace_t() at backtrace.cc:203 (discriminator 1)
       2: format_backtrace(bool) at backtrace.cc:282
       3: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       4: unittest::count_evals(unittest::test_rdb_env_t*, ql::protob_t<Term const>, unsigned int*, unittest::verify_callback_t*) at rdb_interruptor.cc:81 (discriminator 2)
       5: std::_Function_handler<void (), std::_Bind<void (*(unittest::test_rdb_env_t*, ql::protob_t<Term const>, unsigned int*, unittest::exists_verify_callback_t*))(unittest::test_rdb_env_t*, ql::protob_t<Term const>, unsigned int*, unittest::verify_callback_t*)> >::_M_invoke(std::_Any_data const&) at counted.hpp:60
       6: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       7: coro_t::run() at coroutines.cc:214
Previous message may not have been written to the log file (cannot open or find log file).
error: Exiting.
Previous message may not have been written to the log file (cannot open or find log file).
/startdir/PKGBUILD: line 46: 14310 Trace/breakpoint trap   (core dumped) ./build/release/rethinkdb-unittest --gtest_filter=-RDBBtree.*
```

**Comment by danielmewes** [comment]
A fix is up in review 2814

**Comment by danielmewes** [comment]
Fixed in v2.0.x 3cb2410be71465e296f3866144caf68302dce9f9 and next 928d8c6
